### ISSUE TITLE
Failure to uncompress on iPXE booted arm64

### ISSUE THREAD

**Issue by rn** [linuxkit/linuxkit#2415]
When booting an arm64 machine via iPXE using a compressed initrd the following message can be found in `dmesg`:
```
[    5.172247] Unpacking initramfs...
[    6.199089] Initramfs unpacking failed: uncompression error
```
Exactly the same `initrd` works fine when booting in `qemu`

**Comment by rn** [comment]
@arm64b @vielmetti any idea what might be going on here? I suspect it might be iPXE related, though that's a bit weird

**Comment by vielmetti** [comment]
@rn I'm looking at this bug report "Fails to load compressed kernels on arm64" https://bugs.launchpad.net/ubuntu/+source/kexec-tools/+bug/1661363

which looks related? Worth a read in any case.

**Comment by rn** [comment]
yeah, though that is with compressed kernels not compressed `initrd`s. I currently also have to use a un-compressed kernel, so that might be related but it's the kernel which will uncompress the initrd and the kernel config has `RD_GZIP` enabled so should be able to handle it (in fact it does it when booting on `qemu`). It might be related to where iPXE places the `initrd` in memory...

**Comment by arm64b** [comment]
@rn since the same initrd can work with QEMU, seems it's related with the firmware instead of the kernel unpack method, probably the FW load the compressed inird into wrong position or the data loaded is not correct...

**Comment by mor1** [comment]
Hm. Trying to remember-- ISTR I did have something like that happen when I first tried, but based on my failing memory and the artefacts and [iPXE boot script](http://www.cl.cam.ac.uk/~rmm1002/t/moby-ipxe.txt) that I currently have in place, it would appear that I didn't replicate this problem. I can try booting the images I have again if that would help? (and we have a spare packet.net machine slot?)

**Comment by rn** [comment]
@mor1 can you point me at your iPXE script? The one I use is [here](https://github.com/linuxkit/linuxkit/blob/master/src/cmd/linuxkit/run_packet.go#L128). If you want to test on a packet machine, try spinning one up. AMS1 sometimes seems low and I have used ewr1 as well. Alternatively, I currently have one which I can give you access to (ping me on slack)

**Comment by mor1** [comment]
@rn sorry was travelling. iPXE script is linked in the comment to which you replied. <http://www.cl.cam.ac.uk/~rmm1002/t/moby-ipxe.txt>

**Comment by justincormack** [comment]
arm64 kernel apparently has no compression support, possibly it also doesnt for initramfs either.
